cpp/src/arrow/compute/kernels/vector_sort_benchmark.cc

```diff
@@ -301,5 +333,38 @@ BENCHMARK(TableSortIndicesInt64Wide)
     })
     ->Unit(benchmark::TimeUnit::kNanosecond);
 
+//
+// Rank benchmark declarations
+//
+
+void ArrayRankSetArgs(benchmark::internal::Benchmark* bench) {
+  // 3 benchmark arguments: size, inverse null proportion, rank tiebreaker
+  bench->Unit(benchmark::kNanosecond);
```

Comment:
Mostly just a question but is there any particular reason to pin this?  I believe `kNanosecond` is the default and setting it this way prevents the user from adjusting it on the command line.

Reply:
Hmm... I blindly reproduced the existing options for SortIndices benchmarks...

But now I see that it was probably because `RegressionSetArgs`, for whatever reason, forces another unit.
Perhaps we should simply remove all these directives. As you point out, they are actually user-hostile while not bringing any obvious value on their own.

Reply:
Ah, I didn't see the setting in `RegrssionSetArgs`.  I don't think it's too big of a deal but agree it might be nice to let these drift and people can pin it all if they care.

That being said, if we back this out, it would probably disturb continuous benchmarking efforts so maybe not worth it.

Reply:
I would hope that continuous benchmarking efforts are keyed on meaningful metrics such as bytes/sec or items/sec, at least where they exist (and IIRC they do).